### PR TITLE
PR#1279 commits approved by hugovdm beyond sffc's review

### PR DESCRIPTION
**Pull request by hugovdm**
This PR shows changes by younies that I approved for unicode-org/icu.

Since I don't yet feel qualified to give the final word on Java-specific things for upstream ICU, I created this PR to show these changes to Shane for a sanity check!

### CODE REVIEW

File: icu4j/main/classes/core/src/com/ibm/icu/impl/units/ComplexUnitsConverter.java [hunk 1]
```diff
@@ -115,6 +115,9 @@ public boolean greaterThanOrEqual(BigDecimal quantity, BigDecimal limit) {
                 // Keep the residual of the quantity.
                 //   For example: `3.6 feet`, keep only `0.6 feet`
                 quantity = quantity.subtract(newQuantity);
+                if (quantity.compareTo(BigDecimal.ZERO) == -1) {
+                    quantity = BigDecimal.ZERO;
+                }
```
Comment (hugovdm) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/units/ComplexUnitsConverter.java:120]: For the future: given the C++ version zeroes not just negative numbers but also "+epsilon", I want to see if I can write a unit test which will output something that's just a smidge above zero.

            // Keep the residual of the quantity.
            //   For example: `3.6 feet`, keep only `0.6 feet`
            //
            // When the calculation is near enough +/- DBL_EPSILON, we round to
            // zero. (We also ensure no negative values here.)
            if ((quantity - roundedQuantity) / quantity < DBL_EPSILON) {
                quantity = 0;
            } else {
                quantity -= roundedQuantity;
            }

File: icu4j/main/classes/core/src/com/ibm/icu/impl/units/ComplexUnitsConverter.java [hunk 2]
```diff
@@ -23,8 +23,8 @@
  * instances of the `UnitConverter` to perform the conversion.
  */
 public class ComplexUnitsConverter {
-    private static final BigDecimal EPSILON = BigDecimal.valueOf(Math.ulp(1.0));
-    private static final BigDecimal EPSILON_MULTIPLIER = BigDecimal.valueOf(1).add(EPSILON);
+    public static final BigDecimal EPSILON = BigDecimal.valueOf(Math.ulp(1.0));
+    public static final BigDecimal EPSILON_MULTIPLIER = BigDecimal.valueOf(1).add(EPSILON);
```
Comment (hugovdm) [inline icu4j/main/classes/core/src/com/ibm/icu/impl/units/ComplexUnitsConverter.java:27]: @sffc :
These are public because they're used from:
src/com/ibm/icu/dev/test/impl/UnitsTest.java
I don't know whether that really necessitates having these public members?
